src/test/java/org/opensearch/security/action/apitokens/ApiTokenAuthenticatorTest.java

```diff
@@ -0,0 +1,194 @@
+
+import org.apache.logging.log4j.Logger;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.filter.SecurityRequest;
+import org.opensearch.security.http.ApiTokenAuthenticator;
+import org.opensearch.security.user.AuthCredentials;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ApiTokenAuthenticatorTest {
+
+    private ApiTokenAuthenticator authenticator;
+    @Mock
+    private Logger log;
+
+    private ThreadContext threadcontext;
+
+    @Before
+    public void setUp() {
+        Settings settings = Settings.builder()
+            .put("enabled", "true")
+            .put("signing_key", "U3VwZXJTZWNyZXRLZXlUaGF0SXNFeGFjdGx5NjRCeXRlc0xvbmdBbmRXaWxsV29ya1dpdGhIUzUxMkFsZ29yaXRobSEhCgo=")
+            .put("encryption_key", "MTIzNDU2Nzg5MDEyMzQ1Ng==")
+            .build();
+
+        authenticator = new ApiTokenAuthenticator(settings, "opensearch-cluster");
+        authenticator.log = log;
+        when(log.isDebugEnabled()).thenReturn(true);
+        threadcontext = new ThreadContext(Settings.EMPTY);
+    }
+
+    @Test
+    public void testAuthenticationFailsWhenJtiNotInCache() {
+        String testJti = "test-jti-not-in-cache";
+        ApiTokenIndexListenerCache cache = ApiTokenIndexListenerCache.getInstance();
+        assertFalse(cache.getJtis().containsKey(testJti));
+
+        SecurityRequest request = mock(SecurityRequest.class);
+        when(request.header("Authorization")).thenReturn("Bearer " + testJti);
+        when(request.path()).thenReturn("/test");
+
+        AuthCredentials credentials = authenticator.extractCredentials(request, threadcontext);
+
+        assertNull("Should return null when JTI is not in allowlist cache", credentials);
+    }
+
+    @Test
+    public void testExtractCredentialsPassWhenJtiInCache() {
+        String testJti =
+            "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJ0ZXN0LXRva2VuIiwiYXVkIjoidGVzdC10b2tlbiIsIm5iZiI6MTczNTMyNjM0NywiaXAiOiJnaGlWTXVnWlBtcHZJMjZIT2hmUTRnaEJ1Qkh0Y2x6c1REYVlxQjVBRklyTkE4SzJCVTdxc2toMVBCOEMzQWpTdVBaREM0THVSM2pjZkdpLzlkU2ZicDBuQTNGMkhtSi9jaDA3cDY2ZWJ6OD0iLCJpc3MiOiJvcGVuc2VhcmNoLWNsdXN0ZXIiLCJleHAiOjIyMTg3NjU2NDMzMCwiaWF0IjoxNzM1MzI2MzQ3LCJjcCI6IjlPS3N4aFRaZ1pmOVN1akprRzFXVWIzVC9RVTZ4YmZTc2hrbllkVUhraHM9In0.kqMSnn5YwhLmeiI_8iIBQ5uhmI52n2MNniAa52Zpfs3TiE_PXKiNbDNs08hNqzGYW772gT7lfvp6kZnFxQ4v2Q";
```

Comment:
Use `Jwts.builder()` to construct jwts for testing: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/auth/BackendRegistry.java#L175-L193